stan/math/prim/arr/functor/coupled_ode_observer.hpp

```diff
@@ -1,38 +1,152 @@
 #ifndef STAN_MATH_PRIM_ARR_FUNCTOR_COUPLED_ODE_OBSERVER_HPP
 #define STAN_MATH_PRIM_ARR_FUNCTOR_COUPLED_ODE_OBSERVER_HPP
 
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/scal/err/check_less.hpp>
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
+#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
+#include <stan/math/prim/scal/meta/broadcast_array.hpp>
+#include <stan/math/prim/arr/fun/sum.hpp>
+#include <stan/math/prim/mat/fun/typedefs.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+// the mat file defines the needed ops_partial_edge for
+// std::vector<var> operands which are needed here
+#include <stan/math/rev/mat/meta/operands_and_partials.hpp>
```

Comment:
`prim` must not bring in `rev`. If you ran `make test-math-dependencies` these should be highlighted.

Reply:
Yeah, I felt bad when doing it... knowing it was wrong. I think I found now a good solution.